implementation/parallel-to-sns-scatter-gather/README.md

```diff
@@ -1,21 +1,25 @@
 
 ## How it works
 
-The code will deploy two stacks:
+As example use case we use a Car Rental search engine(```requester```) that sends a quote request to several Car Rentals(```responder```). The Car Rental search engine returns the aggregated(```aggregator```) results from multiple suppliers to the user.
+
+The code deploys two stacks:
 
 1. ScatterGatherWithParallelStack
 2. ScatterGatherWithSNSStack
 
-Both stacks deploy 3 lambda functions (application code is re-used for both stacks), ```requester```, ```responder``` and ```aggregator```. The ```requester``` sends a request for quote message providing it with a ```uuid```, multiple ```responder``` will processes the message and outputs a modified message containing a ```price_quote``` specific to the vendor (provided as config parameter in ```cdk.json``` file). The ```aggregator``` aggregates the received quotes.
+Both stacks deploy 3 lambda functions (application code is re-used for both stacks), ```requester```, ```responder``` and ```aggregator```. The ```requester``` sends a request for quote message with an ```uuid```. Multiple ```responder```(Car Rental) process the message modifying the message adding a ```price_quote```. The Car Rentals configuration parameters are defined in the ```cdk.json``` file. Ultimately, the ```aggregator``` function aggregates the received quotes.
 
-ScatterGatherWithParallelStack deploys a Step Function workflow using the parallel state to send a quote request 2 ```responder```. Each ```responder``` adds a price_quote to the original message. When all parallel executions have completed the result is aggregated by the Step Function parallel state.
+ScatterGatherWithParallelStack deploys a Step Function workflow using the parallel state to send a quote request to ```responder```. Each ```responder``` adds a price_quote to the original message. When all parallel executions have completed the result is aggregated by the Step Function parallel state.
 
-ScatterGatherWithSNSStack replaces the parallel state with SNS. The ```requester``` function sends a quote request to an SNS topic. The SNS topic is subscribed by 2 ```responder``` Lambda functions which add a price_quote and send the message to an SQS to aggregate the messages. The ```aggregator``` lambda function synchronously polls the SQS queue. No aggregation logic is provided but the ```uuid``` can be used to determine the quote request the price quote belongs.
+ScatterGatherWithSNSStack replaces the parallel state with SNS. The ```requester``` function sends a quote request to an SNS topic. The SNS topic is subscribed by 2 ```responder``` Lambda functions which add a price_quote and send the message to an SQS to aggregate the messages. The ```aggregator``` lambda function synchronously polls the SQS queue. In order to provide the same aggregation capability as the Step Functions parallel state the ```aggregator``` uses a DynamoDB table to aggregate the price_quotes based on the ```uuid```.
```

Comment:
How does the requestor know when all responses are in? will the workflow time or out or get a callback?

Reply:
I left it out of the solution. Since it is choreography it is difficult to have a callback. I would leave it out. The "car rental search engine" would handle it based on defined wait for responses. It has also all information to check if al vendors responded.